types/react-codemirror/index.d.ts

```diff
@@ -11,7 +11,6 @@
 declare namespace ReactCodeMirror {
     interface ReactCodeMirrorProps {
         children?: React.ReactNode;
-        ref?: React.LegacyRef<ReactCodeMirror> | undefined;
```

Comment:
Props can't determine the ref type for class components. Needed to move code around so that RCM correctly extends a React.Component.